### DISCUSSION TITLE
Regexp to extract facebook username from the url to create a custom title

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#106] [Q&A] [answered]
Based on [this discussion](https://github.com/buggyj/tiddlyclip-plugin/discussions/105) regarding twitter username extraction, I want to create a titlerule for facebook.
But in case of facebook, the variable after the username changes unlike in case of twitter.

Check the below examples .
### Facebook

<img width="1048" alt="image" src="https://user-images.githubusercontent.com/67494083/197062794-c00b8e80-c6e4-473b-8066-8c3be23abd9b.png">

<img width="983" alt="image" src="https://user-images.githubusercontent.com/67494083/197062852-54d32025-1164-4e71-bea3-ac9ecd24555b.png">

### Twitter

<img width="1000" alt="image" src="https://user-images.githubusercontent.com/67494083/197063230-3b76e9f1-86c6-488b-86c1-f64f988507c3.png">

So how to modify the regexp to include only the username in the tiddler title (example urls are given below)
`https://www.facebook.com/curlytales/` `https://www.facebook.com/curlytales/photos/pcb.5822087674507549/5822087551174228/`

**Comment by buggyj** [comment]
Try changing the rule to 

`#rule=/https:\/\/www.facebook.com\/(.*?)\/.*$/g`

**Comment by arunnbabu81** [comment]
> <img alt="image" width="1048" src="https://user-images.githubusercontent.com/67494083/197062794-c00b8e80-c6e4-473b-8066-8c3be23abd9b.png">

It doesnt work for this url - `https://www.facebook.com/curlytales` See the tiddler created after clipping.
<img width="881" alt="image" src="https://user-images.githubusercontent.com/67494083/197206124-3773da55-9423-4adf-974c-3aa1e0e2458c.png">


> <img alt="image" width="983" src="https://user-images.githubusercontent.com/67494083/197062852-54d32025-1164-4e71-bea3-ac9ecd24555b.png">
But it works for this url - `https://www.facebook.com/curlytales/photos/pcb.5822087674507549/5822087551174228/` See the tiddler created after clipping.

<img width="869" alt="image" src="https://user-images.githubusercontent.com/67494083/197206348-e7bc1958-2db5-4043-ab07-c8deb5e01a0f.png">

**Comment by buggyj** [comment, marked as answer]
try this

```
#rule=/https:\/\/www.facebook.com\/(.*?)\/.*$/g
#term=$1
#res=((*@pageRef/#rule/#term*))
#rule=/https:\/\/www.facebook.com\/(.*?)$/g
$title=Facebook/((*#res/#rule/#term*))/((*@clip*))
```

**Reply by arunnbabu81** [reply]
That one works. Thank you @buggyj
